db/schema.rb

```diff
@@ -193,6 +193,8 @@
     t.boolean  "approved",               default: false
     t.string   "github_access_token"
     t.boolean  "admin",                  default: false
+    t.string   "provider"
+    t.string   "uid"
```

Comment:
I have long thought about it.
There are two ways: 
Support a single provider for each user;
Add has_many association for User model and use belongs_to for each provider (and create Provider model);

But it seems that most of users will only use a single account.